.github/workflows/zfs-linux.yml

```
@@ -52,20 +47,18 @@ jobs:
     - uses: actions/download-artifact@v3
     - name: Generating summary
       run: |
-        tar xzf scripts/scripts.tgz .github tests
+        tar xzf modules/modules-22.04.tgz .github tests
```

${{ matrix.os }} won't work here? Just a nit
```suggestion
        tar xzf modules/modules-${{ matrix.os }}.tgz .github tests
```

---

Yes, ${{ matrix.os }}.tgz will not work there, because the variable is not set. The cleanup job in the end downloads all generated artifacts and uses `runs-on: ubuntu-22.04`.